### PR TITLE
Tempus: Use Clang-format

### LINKED ISSUE #12659: Tempus: Use Clang-format
## Enhancement
Following MueLu's lead, Tempus will start using clang-format.

Tempus will use the Google base style, the Kokkos options, and
Tempus-specific options.

### PR DESCRIPTION
**Pull request by ccober6**
Following MueLu's lead, Tempus will start using clang-format.

Tempus will use the Google base style, the Kokkos options, and Tempus-specific options.

@trilinos/tempus 

## Related Issues

* Closes #12659

### CODE REVIEW

File: .github/workflows/clang_format.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
     - uses: actions/checkout@v3
     - uses: DoozyX/clang-format-lint-action@v0.16.2
       with:
-        source: './packages/muelu'
+        source: './packages/muelu ./packages/tempus'
         exclude: ''
```
Comment (cgcgcg) [inline .github/workflows/clang_format.yml:14]: @ccober6 : Change to
```
exclude: ./packages/tempus/examples
```

Comment (ccober6) [reply .github/workflows/clang_format.yml]: Done.  Thanks!

**Review by cgcgcg — DISMISSED** [review]

File: packages/tempus/unit_test/Tempus_UnitTest_SolutionHistory.cpp [hunk 2]
```diff
@@ -32,78 +30,84 @@ TEUCHOS_UNIT_TEST(SolutionHistory, Default_Construction)
...
 // ************************************************************
 TEUCHOS_UNIT_TEST(SolutionHistory, Full_Construction)
 {
   std::string name = "Unit Test";
 
-  auto history = rcp(new std::vector<RCP<Tempus::SolutionState<double> > >);;
-  auto model = rcp(new Tempus_Test::SinCosModel<double>());
+  auto history = rcp(new std::vector<RCP<Tempus::SolutionState<double> > >);
+  ;
+  auto model    = rcp(new Tempus_Test::SinCosModel<double>());
```
Comment (rppawlo) [inline packages/tempus/unit_test/Tempus_UnitTest_SolutionHistory.cpp:78]: this probably is not what you want clang format to do for extra semicolon

Comment (ccober6) [reply packages/tempus/unit_test/Tempus_UnitTest_SolutionHistory.cpp]: Interesting ... There were about a dozen places where we had double semicolons (';;').  Not sure it is something that clang-format can handle "correctly" for us.  It is a benign statement, but clang-format at least pointed it out.  Removing them.   Can it get an approval?

**Review by rppawlo — APPROVED** [review]

**Pull request merged**